include/swift/SILOptimizer/Analysis/ARCAnalysis.h

```diff
@@ -386,4 +390,179 @@ SILInstruction *findReleaseToMatchUnsafeGuaranteedValue(
 
 } // end namespace swift
 
+
+namespace swift {
+
+/// EpilogueARCBlockState - Keep track of whether a epilogue ARC instruction has
+/// been found.
+struct EpilogueARCBlockState {
+  /// Keep track of whether a eplogue release has been found before and after
+  /// this basic block.
+  bool BBSetIn;
+  bool BBSetOut;
+  /// Constructor, we only compute epilogue ARC instruction for 1 argument at
+  /// a time.
+  EpilogueARCBlockState() {
+    // Optimistic data flow.
+    BBSetIn = true;
+    BBSetOut = false;
```

Comment:
Do you really need to store BSSetOut?
Isn't it sufficient to calculate it in a local variable in convergeDataflow() and computeEpilogueARC()?